### ISSUE TITLE
Reorder columnIndex on initialization?

### ISSUE THREAD

**Issue by LabShareLouie** [TonyGermaneri/canvas-datagrid#133]
**Expected behavior:** On grid initialization order by user defined columnIndex.

 **Actual behavior:** Unknown solution. No examples in documentation.

@TonyGermaneri please advise if user init column ordering is possible... If so is there an example?

Thanks for dev'ing the grid. Very much appreciate the module!

-- Louie

**Comment by TonyGermaneri** [comment]
I think this is what you're looking for:

https://tonygermaneri.github.io/canvas-datagrid/docs/#tutorial--Order-by-a-column

In this example, it orders by column index 0.  If you knew the name of the column you could just pass the name instead of resolving the name using the schema.

Is this what you were looking for?

**Comment by LabShareLouie** [comment]
Unfortunately not. The example orders by **a** column. I wish to order **the** columns.

For example instead of column order [A (col 1), B (col 2), C (col 3), D (col 4)], I would like to initialize it like so:  [ B (col 2), [A (col 1), D (col 4),  C (col 3) ]

**Comment by LabShareLouie** [comment]
A practical use case is if a user drags the headers to a certain columns-order. This order can somehow be saved, so when the user refreshes, the columns-order stays the same.

**Comment by TonyGermaneri** [comment]
Ah, I understand.  There is no way to do it right now.  But it's easy to implement, so I'll add the feature in the next release.

The ordering of the columns internally is an array of integers representing the index of the column in the schema.  So to reverse the order of a 3 column grid you would have an array `[2, 1, 0]`.

I'll expose the property via a getter/setter and add a bit of validation (length and type checks).

**Comment by TonyGermaneri** [comment]
> A practical use case is if a user drags the headers to a certain columns-order. This order can somehow be saved, so when the user refreshes, the columns-order stays the same.

Sorry I just really read that comment and let it sink in what you were saying is your use case for the first time.  If you pass the grid a name during instantiation it will by default save the orders of the columns (if the ability to reorder columns is enabled) to local storage on the browser using the name you provided.

I added the property you requested.  I also added the row version of the same as strange as that is.  It's merged in to the master branch and will be in 0.20.19 when it comes out.
